app/src/main/java/net/grandcentrix/blutufintegration/ui/launch/LaunchFragment.kt

```diff
@@ -46,53 +40,35 @@ class LaunchFragment : Fragment() {
         (requireActivity() as AppCompatActivity).supportActionBar?.hide()
         binding = FragmentLaunchBinding.inflate(layoutInflater, container, false)
 
-        /*val enableLocationIntent = Intent(BluetoothAdapter.ACTION_REQUEST_DISCOVERABLE)
-        startActivityForResult(enableLocationIntent, 2)*/
-
-        launch()
-
         return binding.root
     }
 
-    private fun launch() {
-        if (anyRequirementsNeeded()) {
-            when {
-                !binding.checkboxBle.isChecked -> showError()
-                !binding.checkboxPermission.isChecked -> updateStartBtn(true)
-            }
-        } else {
-            updateStartBtn(false)
-        }
-    }
-
-    private fun anyRequirementsNeeded(): Boolean = listOf(
-        isBluetoothSupported(),
-        checkPermissions()
-    ).any { checkResult -> checkResult == CheckResult.REQUIRED }
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
 
-    @VisibleForTesting
-    fun isBluetoothSupported(): CheckResult {
-        val result = when (repository.deviceSupportsBluetooth(requireContext())) {
-            true -> CheckResult.PASSED
-            else -> CheckResult.REQUIRED
-        }
-        binding.checkboxBle.isChecked = result == CheckResult.PASSED
-        return result
+        viewModel.conditions.observe(viewLifecycleOwner) { conditions -> updateUi(conditions) }
     }
 
-    @VisibleForTesting
-    fun checkPermissions(): CheckResult {
-        val result = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M &&
-            permissions.any {
-                checkSelfPermission(requireContext(), it) !=
-                        PackageManager.PERMISSION_GRANTED
+    private fun updateUi(conditions: List<MissingPrecondition>) {
+        Log.e("sami", "error: ${conditions.size}")
+
+        binding.checkboxPermission.isChecked = true
```

Comment:
Setting the `Checkbox` state here without any condition but later setting the value with a condition is double work.
```suggestion
        binding.checkboxPermission.isChecked = conditions.all { cond -> cond !is FineLocationPermissionMissingError }
```